src/main/java/com/ldtteam/structurize/client/gui/WindowBuildTool.java

```diff
@@ -881,6 +886,28 @@ private void changeSchematic()
         {
             Settings.instance.setPosition(this.pos);
         }
+
+        adjustToGroundOffset();
```

Comment:
If the ground level doesn't change between the levels then there will be no change, it subtracts the previous offset and then adds the new offset, resulting in a net zero difference in the position if they were the same.  This will work regardless of whether they're switching between levels or switching between different schematics that happen to be the same.